### PR TITLE
Add StackArray example

### PR DESCRIPTION
**Pull request by guilhermeleobas**
This PR adds an StackArray to the list of example as a complement on how to use some Numba APIs.

### CODE REVIEW

File: examples/stackarray.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+@overload(operator.getitem)
+def stackarray_getitem(a, i):
+    if isinstance(a, StackArrayType):
+        def impl(a, i):
+            return a.ptr[i]
+        return impl
+
+
+@overload(len)
+def omnisci_array_len(x):
```
Comment (seibert) [inline examples/stackarray.py:93]: Did you mean to name this `stackarray_len`?  :)

**Comment by rgommers** [comment]
Nice @guilhermeleobas. Maybe would be useful to add a docstring with some motivation/explanation? E.g. "if Python overhead of an array is a bottleneck in your code, such as when you have an array of indices that you need to use in a tight loop, then `StackArray` shows a way of pre-allocating an array on the stack to avoid that overhead. ..... `ArrayStack`, `ArrayStackType` and `ArrayStackModel` are <add what they do / how they relate - I don't fully understand that part from a quick read through>)".

**Comment by esc** [comment]
@guilhermeleobas what was the conclusion on this one? Will it be submitted to `numba-examples` instead?

**Comment by stuartarchibald** [comment]
> @guilhermeleobas what was the conclusion on this one? Will it be submitted to `numba-examples` instead?

From core dev meeting, yes. `numba-examples` needs a `mkdir` doing for `user_contributed` or similar as a location to put all these.

**Comment by guilhermeleobas** [comment]
Yeah, I will open a new PR in the `numba-examples` with this.

**Comment by guilhermeleobas** [comment]
Is the `numba-examples` more related to performance? In the PR, should I adapt the example to look more like a benchmark?

**Pull request closed without merging**